src/lib/queue_worker.js

```diff
@@ -171,18 +171,24 @@ QueueWorker.prototype._resolve = function(taskNumber) {
         var id = self.processId + ':' + self.taskNumber;
         if (task._state === self.inProgressState &&
             task._owner === id) {
-          if (_.isNull(self.finishedState)) {
-            return null;
+          var outputTask = _.clone(newTask);
+          if (!_.isPlainObject(outputTask)) {
+            outputTask = {};
           }
-          if (!_.isPlainObject(newTask)) {
-            newTask = {};
+          outputTask._state = _.get(outputTask, '_new_state');
+          delete outputTask._new_state;
+          if (!_.isNull(outputTask._state) && !_.isString(outputTask._state)) {
+            if (_.isNull(self.finishedState) || outputTask._state === false) {
```

Comment:
Can you add a comment about the explicit `false` API just because it's not necessarily intuitive.